src/pharmpy/tools/mfl/parse.py

```diff
@@ -309,6 +315,38 @@ def replace(self, **kwargs):
             metabolite=metabolite,
         )
 
+    def replace_features(self, mfl_str):
+        key_dict = {
+            Absorption: 'absorption',
+            Elimination: 'elimination',
+            Transits: 'transits',
+            Peripherals: 'peripherals',
+            LagTime: 'lagtime',
+            Covariate: 'covariate',
+            DirectEffect: 'direct_effect',
+            EffectComp: 'effect_comp',
+            IndirectEffect: 'indirect_effect',
+            Metabolite: 'metabolite',
+        }
+        mfl_list = _parse(mfl_str)
+        kwargs = {}
+        for statement in mfl_list:
+            key = key_dict[type(statement)]
+            if type(statement) in (Transits, Peripherals, Covariate, IndirectEffect):
+                no_of_modes = len(statement.counts)
```

Comment:
I believe the `Covariate` statements do not have the `.counts` attribute. If the number of covariate effects it represents wants to be used, I would check the combinations of parameter/covariate